src/lib/utils/os_utils.cpp

```diff
@@ -163,6 +167,27 @@ uint64_t OS::get_cpu_cycle_counter()
    return rtc;
    }
 
+long OS::get_cpu_total()
+   {
+   static long tt = 0;
```

Comment:
Remove the caching, this will cause warnings from race checkers since the access is not serialized. I know the call is slightly expensive but usually it will be followed by spawning some threads which is far more costly, so I don't think the overhead matters.